From hidave.darkstar@gmail.com  Tue Feb  3 16:22:36 2009
From: Dave Young <hidave.darkstar@gmail.com>
Date: Sun, 01 Feb 2009 18:54:54 +0800
Subject: USB: usb-serial: fix the aircable_init failure path
To: wgreathouse@smva.com, linux-usb@vger.kernel.org, linux-kernel@vger.kernel.org
Message-ID: <49857F7E.9090708@gmail.com>


The failure path of aircable_init is wrong, fix the order of (goto) labels.

Signed-off-by: Dave Young <hidave.darkstar@gmail.com>
Acked-by: Naranjo Manuel Francisco <naranjo.manuel@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/aircable.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/serial/aircable.c
+++ b/drivers/usb/serial/aircable.c
@@ -621,9 +621,9 @@ static int __init aircable_init(void)
 		goto failed_usb_register;
 	return 0;
 
-failed_serial_register:
-	usb_serial_deregister(&aircable_device);
 failed_usb_register:
+	usb_serial_deregister(&aircable_device);
+failed_serial_register:
 	return retval;
 }
 
